app/actions/websocket/posts.ts

```diff
@@ -217,5 +215,5 @@ export async function handlePostUnread(serverUrl: string, msg: WebSocketMessage)
     const channel = channels?.[0];
     const postNumber = channel?.total_msg_count;
     const delta = postNumber ? postNumber - msg.data.msg_count : msg.data.msg_count;
-    markChannelAsUnread(serverUrl, msg.broadcast.channel_id, delta, msg.data.mention_count, true, msg.data.last_viewed_at);
+    markChannelAsUnread(serverUrl, msg.broadcast.channel_id, delta, msg.data.mention_count, msg.data.last_viewed_at);
```

Comment:
This may be problematic. We are setting the channel as unread because a websocket event. This function (as far as I know) does not distinguish whether it should send a message to the server or not, so will always send a message to the server. This will generate another websocket event, that hopefully won't be sent to this client. But if we have two mobile clients connected at the same time, they will be ping-pong'ing the message, creating issues. We have to determine whether this come from websocket or not to actually send the message to the server or not.

Reply:
Isn't this calling the local action that does not execute a request?

Reply:
My bad. You are completely right.